Formula/fb303.rb

```diff
@@ -51,10 +51,10 @@ def install
         return 0;
       }
     EOS
-    system ENV.cxx, "-std=c++17", "-I#{include}", "-I#{Formula["openssl@1.1"].opt_include}", "test.cpp",
+    system ENV.cxx, *ENV.cxxflags.split, "-std=c++17", "-I#{include}", "-I#{Formula["openssl@1.1"].opt_include}",
```

Comment:
This fixes the test on macOS but breaks it on Linux...

Reply:
`test.cpp` needs to be before the linker flags.

Reply:
Ah right

Reply:
Looks like `-march=native` does still cause issues on Linux as there are now other libraries that need to be linked.

Interesting that `ENV.cxxflags` has `-march=nehalem` on macOS. I guess difference in behavior could be from Linux shims like https://github.com/Homebrew/brew/blob/master/Library/Homebrew/extend/os/linux/extend/ENV/shared.rb#L12

In theory, the more accurate compilation on Linux would be adding same build-time flags too, but excluding them doesn't cause any issues for now.

Reply:
Looks like issue isn't just `-march=native` (which does break `folly` linkage) but also `-Os` optimization.

If we want to leave that, then can try adding `-lthriftcpp2`.